postmortems/2024-03-22_azure_dev_outage.md

```diff
@@ -0,0 +1,110 @@
+# 2024-03-22 Azure dev deployment week long outage
+
+## What
+
+The app registration secret in client's Azure deploy expired on 22 March 2024, which caused the authorisation flow for the App to be unable to access the Azure Active Directory to authenticate users, and meant that users were unable to access the platform. 
+
+After we became aware the secret had expired, the secret was not able to be regenerated internally.  
+
+Once the secret was regenerated by a third-party supplier, governance processes had to be followed to reinstate the secret. 
+
+
+2024 (all times UTC)
+* 2024/03/22 Friday 13:17 – app registration Secret expired 
+* 2024/03/22 Friday 14:38 – First request to authorisation server with a denial; generated an Azure Monitor Alert sent to Support Inbox 
+* 2024/03/25 Monday 10:38 – First Azure Monitor Alert forwarded to MP & AL (developers)
+* 2024/03/25 Monday 13:29 Development Team Sprint Planning session begins; Product Owner (PO), QA, and all developers are among attendees – this is only relevant as an indication that the team were in a meeting 
+* 2024/03/25 Monday 13:46 Internal user reports trying to access the site and being unable to log in 
+* 2024/03/25 Monday 13:47 Confirmed that second internal user is also not able to log in 
+* 2024/03/25 Monday 13:54 MP confirms that the app registration secret for the prod version of the app has expired and needs to be regenerated 
+* 2024/03/25 Monday 13:57 MP discovers his account for client's Azure env is blocked 
+* 2024/03/25 Monday 14:02 MP is reminded that the guest B2B account is unable to regenerate secrets, despite being an owner of the app registration for which the secret has expired
+* 2024/03/25 Monday 15:04 Development Team Sprint Planning ends 
+* 2024/03/25 Monday 15:19 MP calls IT Service Desk, account is blocked because it has expired and needs account extension from line manager for the password to be reset and the account reactivated. Expired due to original timing authorized for account.  
+* 2024/03/25 Monday 15:48 Line manager who must request the account extension is identified (normal LM is on long term leave) 
+* 2024/03/25 Monday 15:53 Notice added to client's Sharepoint indicating that app is down 
+* 2024/03/25 Monday 16:48 Build Pipelines and Azure Repo are discovered to be not visible to MP & AL who previous could see them – reason not clear at this stage 
```

Comment:
This was annoying and something that we did discover at the same time while we were debugging permissions, but I don't think it's relevant to the app outage?

Reply:
I think it contributes to the overall picture that there are unexpected things happening that pose a risk, and that demonstrate failure on both sides. Happy to remove though if you don't feel this is necessary. Shall I remove @alightwing?

Reply:
Changes made